print-icmp.c

```diff
@@ -782,139 +918,12 @@ icmp_print(netdissect_options *ndo, const u_char *bp, u_int plen,
-                        switch (GET_BE_U_2(ipaddr_subobj->afi)) {
-                            case 1:
-                                ND_PRINT("%s", GET_IPADDR_STRING(ipaddr_subobj->ip_addr));
-                                offset += 4;
-                                break;
-                            case 2:
-                                ND_PRINT("%s", GET_IP6ADDR_STRING(ipaddr_subobj->ip_addr));
-                                offset += 16;
-                                break;
-                            default:
-                                ND_PRINT("Unknown Address Family Identifier");
-                                return;
-                        }
-                        offset += 4;
-                    }
-                    if (name_flag) {
-                        uint8_t inft_name_length_field;
-                        const struct icmp_interface_information_ifname_subobject_t *ifname_subobj;
-
-                        ifname_subobj = (const struct icmp_interface_information_ifname_subobject_t *) offset;
-                        inft_name_length_field = GET_U_1(ifname_subobj->length);
-                        ND_PRINT("\n\t    Interface Name");
-                        if (inft_name_length_field == 0) {
-                            ND_PRINT(" [length %u]", inft_name_length_field);
-                            nd_print_invalid(ndo);
-                            break;
-                        }
-                        if (inft_name_length_field % 4 != 0) {
-                            ND_PRINT(" [length %u != N x 4]", inft_name_length_field);
-                            nd_print_invalid(ndo);
-                            offset += inft_name_length_field;
-                            break;
-                        }
-                        if (inft_name_length_field > 64) {
-                            ND_PRINT(" [length %u > 64]", inft_name_length_field);
-                            nd_print_invalid(ndo);
-                            offset += inft_name_length_field;
-                            break;
-                        }
-                        ND_PRINT(", length %u: ", inft_name_length_field);
-                        nd_printjnp(ndo, ifname_subobj->if_name,
-                                    inft_name_length_field - 1);
-                        offset += inft_name_length_field;
-                    }
-                    if (mtu_flag) {
-                        ND_PRINT("\n\t    MTU: %u", GET_BE_U_4(offset));
-                        offset += 4;
-                    }
-                    break;
-                  }
-
-                default:
-                    print_unknown_data(ndo, obj_tptr, "\n\t    ", obj_tlen);
-                    break;
-                }
-                if (hlen < obj_tlen)
+                int obj_tlen = print_icmp_multipart_ext_object(ndo, obj_tptr);
+                if (obj_tlen < 0) {
+		    nd_print_trunc(ndo);
```

Comment:
Because ND_LONGJMP_FROM_TCHECK is defined in this printer, all truncated cases should longjmp(). Thus we should use nd_trunc_longjmp(ndo) here.

Reply:
But are 'return -1' truncated cases or just invalid cases?

Reply:
Thanks for catching this. I will re-review tomorrow.

Reply:
> It's possible that there's an improvement that would continue the loop in the case of an unknown address family, since we know the length of the overall object, but, that can be a future change.

OK.